### ISSUE TITLE
Unreliable crafters and automation

### ISSUE THREAD

**Issue by Omikhleia** [pandorabox-io/pandorabox.io#872]
This once used to work, but no longer?

- Autocrafters are no longer crafting some recipes (from several months ago)

  paper to books...
  ![image](https://github.com/user-attachments/assets/92a98e22-b28f-4521-a257-b7250173ce51)

  purple dye and wool to purple wool...
  ![image](https://github.com/user-attachments/assets/cf2f7429-56a5-47ab-8214-95253b978ff7)

- Water breakers no longer ejecting filled buckets 
  ![image](https://github.com/user-attachments/assets/88e9caab-8dc8-480a-903c-0a79b45fc078)

  (It used to... And the obvious workaround here is to always have empty buckets in the breaker slot, but eventually it may break by mischance)

**Comment by SwissalpS** [comment]
Thanks for reporting. Yes, we are aware.
The book issue is because of [pandorabox_custom] book repair recipe.
The wool is because of an alias (violet <-> purple).

I'm working on improving the autocrafter.

The water bucket in a breaker has happened to me too, have not yet narrowed it down to why exactly. (This should be a separate issue)

**Comment by BuckarooBanzay** [comment]
> Water breakers no longer ejecting filled buckets

This still works for me:

![screenshot_20240811_095946](https://github.com/user-attachments/assets/add75bfb-9479-4dce-a0fb-0b53fd0e8354)

They always ejected on the back-side though :thinking:

**Comment by SwissalpS** [comment]
Yes out the back, but sometimes when there are no empty buckets left, somehow the filled bucket ends up in the inventory. Even if the target node has plenty of space to receive the full bucket.

**Comment by FeXoR-o-Illuria** [comment]
I thought the full water bucket breaking/digging water isn't possible - same as the player did it - so ejection isn't triggered.
Not sure if I consider that a bug.

The water bucket ending up in the breaker in the first place is probably a bug but unrelated in my opinion.
